controllers/cloud.redhat.com/namespacereservation_controller_suite_test.go

```diff
@@ -125,6 +111,16 @@ var _ = Describe("Reservation controller basic reservation", func() {
 				}
 				return false
 			}, timeout, interval).Should(BeTrue())
+
+			By("Creating the number of ready namespaces as the pool size")
+			pool := crd.NamespacePool{}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "minimal"}, &pool)
+				Expect(err).NotTo(HaveOccurred())
+
+				return pool.Spec.Size == pool.Status.Ready
+			}, timeout, interval).Should(BeTrue())
```

Comment:
I think in a followup PR we should do the following 
1) Ensure that we check ALL the reservations we created, not just 2 of them. The logic is still heavily based around one of them being waiting, we should change that.
2) We should ensure the resources actually get created - at the moment the test is asserting the status was updated, but we are not checking that the namespace itself has been updated.

Reply:
Will create a new branch to implement this change after this branch is merged.